dashboard/src/app/[locale]/(onboarding)/onboarding/steps/WebsiteSetup.tsx

```diff
@@ -3,49 +3,74 @@
 import { domainValidation } from '@/entities/dashboard';
 import { toast } from 'sonner';
 import { PrefixInput } from '@/components/inputs/PrefixInput';
-import { useTranslations } from 'next-intl';
+import { useTranslations, useLocale } from 'next-intl';
 import { useOnboarding } from '../OnboardingProvider';
+import { Checkbox } from '@/components/ui/checkbox';
+import { useSession } from 'next-auth/react';
+import { Tooltip, TooltipContent, TooltipProvider, TooltipTrigger } from '@/components/ui/tooltip';
 
 type WebsiteSetupProps = {
   onNext: Dispatch<void>;
+  showTos?: boolean;
 };
 
-export default function WebsiteSetup({ onNext }: WebsiteSetupProps) {
+export default function WebsiteSetup({ onNext, showTos }: WebsiteSetupProps) {
   const t = useTranslations('onboarding.website');
+  const tValidation = useTranslations('validation');
+  const locale = useLocale();
   const [error, setError] = useState('');
   const [isPending, startTransition] = useTransition();
+  const [agree, setAgree] = useState(false);
 
   const { setDashboard } = useOnboarding();
+  const { update } = useSession();
 
-  const handleSubmit = useCallback(async (e: React.FormEvent<HTMLFormElement>) => {
-    e.preventDefault();
-    setError('');
+  const handleSubmit = useCallback(
+    async (e: React.FormEvent<HTMLFormElement>) => {
+      e.preventDefault();
+      setError('');
 
-    const formData = new FormData(e.currentTarget);
-    const domain = formData.get('domain') as string;
-
-    const domainResult = domainValidation.safeParse(domain);
-    if (!domainResult.success) {
-      setError(domainResult.error.errors[0]?.message || 'Invalid domain');
-      return;
-    }
+      if (showTos && !agree) {
+        setError(tValidation('termsOfServiceRequired'));
+        return;
+      }
 
-    startTransition(async () => {
-      const result = await createDashboardAction(domainResult.data);
+      const formData = new FormData(e.currentTarget);
+      const domain = formData.get('domain') as string;
 
-      if (!result.success) {
-        setError(t('failedToCreateDashboard'));
+      const domainResult = domainValidation.safeParse(domain);
+      if (!domainResult.success) {
+        setError(domainResult.error.errors[0]?.message || 'Invalid domain');
         return;
       }
-      setDashboard(result.data);
-      toast.success(t('dashboardCreatedSuccess'));
-      onNext();
-    });
-  }, []);
+
+      startTransition(async () => {
+        const result = await completeOnboardingAndCreateDashboardAction({
+          domain: domainResult.data,
+          acceptTerms: showTos ? agree : undefined,
+          language: locale,
+        });
+
+        if (!result.success) {
+          setError(result.error.message);
+          return;
+        }
+
+        try {
+          await update();
+        } catch {}
+
+        setDashboard(result.data);
+        toast.success(t('dashboardCreatedSuccess'));
+        onNext();
```

Comment:
In case the `update` fails, it should probably not continue, no? 😄 

```suggestion
        try {
          await update();
          setDashboard(result.data);
          toast.success(t('dashboardCreatedSuccess'));
          onNext();
        } catch {
          setError('Please try again');
        }
```